src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php

```diff
@@ -86,17 +86,43 @@ public function renderAttributeValueFormsAction(Request $request): Response
      */
     protected function getAttributeFormsInAllLocales(AttributeInterface $attribute, array $localeCodes): array
     {
-        $attributeForm = $this->get('sylius.form_registry.attribute_type')->get($attribute->getType(), 'default');
+        $attributeForm = $this
+            ->get('sylius.form_registry.attribute_type')
+            ->get($attribute->getType(), 'default')
+        ;
 
         $forms = [];
+
+        if (!$attribute->isTranslatable()) {
+            array_push($localeCodes, null);
+
+            return $this->createFormAndView($attributeForm, $attribute, null, $forms);
+        }
+
         foreach ($localeCodes as $localeCode) {
-            $forms[$localeCode] = $this
-                ->get('form.factory')
-                ->createNamed('value', $attributeForm, null, ['label' => $attribute->getName(), 'configuration' => $attribute->getConfiguration()])
-                ->createView()
-            ;
+            $forms = $this->createFormAndView($attributeForm, $attribute, $localeCode, $forms);
```

Comment:
Why did you extract this to a private method? This extraction changed the current behaviour, is it intentional?